c/src/measureresult.cpp

```diff
@@ -38,33 +38,52 @@ tirexError tirexResultEntryNum(const tirexResult* result, size_t* num) {
 
 void tirexResultFree(tirexResult* result) { delete result; }
 
-template <class... Ts>
-struct overloaded : Ts... {
-	using Ts::operator()...;
-};
+static std::string toYAML(const tirex::TimeSeries<unsigned>& timeseries) {
+	const auto& [timestamps, values] = timeseries.timeseries();
+
+	std::stringstream timestampsStream;
+	timestampsStream << "[";
+	bool firstTImestamp = true;
+	for (auto& elem : timestamps) {
+		if (!firstTImestamp)
+			timestampsStream << ",";
+		timestampsStream << "\"";
+		timestampsStream << elem.count();
+		timestampsStream << "\"";
+		firstTImestamp = false;
+	}
+	timestampsStream << "]";
 
-template <class... Ts>
-overloaded(Ts...) -> overloaded<Ts...>;
+	std::stringstream valuesStream;
+	valuesStream << "[";
+	bool firstValue = true;
+	for (auto& elem : values) {
+		if (!firstValue)
+			valuesStream << ",";
+		valuesStream << "\"";
+		valuesStream << elem;
+		valuesStream << "\"";
+		firstValue = false;
+	}
+	valuesStream << "]";
 
-template <typename T>
-static std::string toYAML(const tirex::TimeSeries<T>& timeseries) {
-	const auto& [timestamps, values] = timeseries.timeseries();
 	return _fmt::format(
-			"{{max: {}, min: {}, avg: {}, timeseries: {{timestamps: [{}], values: [{}]}}}}", timeseries.maxValue(),
-			timeseries.minValue(), timeseries.avgValue(), tirex::utils::join(timestamps, ','),
-			tirex::utils::join(values, ',')
+			"{{\"max\":{},\"min\":{},\"avg\":{},\"timeseries\": {{\"timestamps\": [{}], \"values\": [{}]}}}}",
+			timeseries.maxValue(), timeseries.minValue(), timeseries.avgValue(), timestampsStream.str(),
+			valuesStream.str()
 	);
 }
 
 extern tirexResult_st* tirex::createMsrResultFromStats(tirex::Stats&& stats) {
 	std::vector<std::pair<tirexMeasure, std::string>> result;
 	for (auto&& [key, value] : stats) {
 		std::visit(
-				overloaded{
-						[key, &result](std::string& str) { result.emplace_back(key, std::move(str)); },
-						[key, &result](const tirex::TimeSeries<unsigned>& timeseries) {
-							result.emplace_back(key, toYAML(timeseries));
-						}
+				[key, &result](auto&& arg) {
+					using T = std::decay_t<decltype(arg)>;
+					if constexpr (std::is_same_v<T, std::string>)
+						result.emplace_back(key, std::move(arg));
+					else if constexpr (std::is_same_v<T, tirex::TimeSeries<unsigned>>)
+						result.emplace_back(key, toYAML(arg));
```

Comment:
I don't see the advantage over overloading. Especially since the type is then again checked within the function. This doubles the number of conditions per call and looks entirely odd, wheras overloading is a standard design pattern. If overloading is not preferred, the visit should be made explicit via [`get_if`](https://en.cppreference.com/w/cpp/utility/variant/get_if).